### PR TITLE
Api contract for users/userAvailble/:id route

### PR DESCRIPTION
**Pull request by saishankarmanugula**
image

### CODE REVIEW

File: users/README.md [hunk 1]
```diff
@@ -111,6 +111,22 @@ Returns the specified user.
   - **Code:** 401
     - **Content:** `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
 
+## **GET /users/userAvailable/:id**
+
+Returns the availability of user name.
+
+- **Params**  
+  _Required:_ `id=[string]`
```
Comment (swarajpure) [inline users/README.md:119]: re: `id`

File: users/README.md [hunk 2]
```diff
@@ -111,6 +111,22 @@ Returns the specified user.
   - **Code:** 401
     - **Content:** `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
 
+## **GET /users/userAvailable/:id**
+
+Returns the availability of user name.
```
Comment (swarajpure) [inline users/README.md:116]: Please combine and make it `username`

File: users/README.md [hunk 3]
```diff
@@ -111,6 +111,22 @@ Returns the specified user.
   - **Code:** 401
     - **Content:** `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
 
+## **GET /users/userAvailable/:id**
```
Comment (swarajpure) [inline users/README.md:114]: There should be `username` instead of `id`, right?

File: users/README.md [hunk 4]
```diff
@@ -111,6 +111,22 @@ Returns the specified user.
...
+  _Required:_ `id=[string]`
+- **Body**  
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Success Response:**
+  - **Code:** 200
+    - **Content:** `{ 'userAvailable': <Availability Stauts> }`
```
Comment (swarajpure) [inline users/README.md:128]: Something like `isUsernameAvailable` sounds better to me. What do you say?
Also the value should be <boolean>

**Review by swarajpure — COMMENTED** [review]
Have some doubts, please clarify.

**Review by swarajpure — CHANGES_REQUESTED** [review]
Looks good to me 👍🏻 

Also, if this route is going to require authentication (which I think it would), please add 4xx error response for unauthorized user
Also, kindly add 5xx error for server error response

You can refer contracts for other `users` routes to get idea of the above two error responses.

Thank you 🎉

**Review by swarajpure — APPROVED** [review]
Great! Thank you! 🎉

**Review by ankushdharkar — APPROVED** [review]

**Comment by ankushdharkar** [comment]
This looks good. Thanks @saishankarmanugula!

**Pull request merged**